subsys/bluetooth/host/gatt.c

```diff
@@ -1910,44 +1910,75 @@ static void gatt_indicate_rsp(struct bt_conn *conn, uint8_t err,
 	}
 }
 
-static int gatt_send(struct bt_conn *conn, struct net_buf *buf,
-		     bt_att_func_t func, void *params)
+static struct bt_att_req *gatt_req_alloc(bt_att_func_t func, void *params,
+					 bt_att_write_buf_t write_buf,
+					 uint8_t op,
+					 size_t len)
 {
+	struct bt_att_req *req;
+
+	/* Allocate new request */
+	req = bt_att_req_alloc(BT_ATT_TIMEOUT);
```

Comment:
We might be better off chaning bt_att_req_alloc to take all these parameters intead of having a wrapper on gatt.c that just initializes the structure.

Reply:
The reason why that wasn't possible at the moment was the bt_gatt_indicate. In order to make indications fit we will have to add a stack internal variable in the indication params to hold the resolved handle, since finding the handle is a rather complicated piece of code and should be kept out of the encode function IMO.